maths/extended_gcd.nim

```diff
@@ -0,0 +1,76 @@
+## Extended Euclidean algorithm
+##
+## Provides the implementation of the function extendedGCD*
+
+import std/math
+
+
+func updateCoefficients(t0, t1, q: int): (int, int) =
+  (t1, t0 - q * t1)
+
+
+func euclidIteration(inA, inB: int): (int, int, int) =
+  assert inA >= 0 and inB >= 0
+  var (a, b) = (inA, inB)
+  var (x0, x1) = (1, 0)
+  var (y0, y1) = (0, 1)
+  while b != 0:
+    let q = a div b
+    (x0, x1) = updateCoefficients(x0, x1, q)
+    (y0, y1) = updateCoefficients(y0, y1, q)
+    (a, b) = (b, a mod b)
+
+  (a, x0, y0)
+
+
+func extendedGCD*(a, b: int): (int, int, int) =
+  ## Implements the
+  ## [Extended Euclidean algorithm](https://en.wikipedia.org/wiki/Extended_Euclidean_algorithm).
+  ## For given integers `a`, `b` it
+  ## computes their [`gcd`](https://en.wikipedia.org/wiki/Greatest_common_divisor)
+  ## and integers `x` and `y`, such that
+  ## `gcd = a * x + b * y`
```

Comment:
In which range lie the Bézout's coefficients that you return?

Reply:
cf. fdfebe6d6a726148a3985c6606735cd9fb304c53